hardware/arduino/avr/cores/arduino/WString.cpp

```diff
@@ -43,16 +43,15 @@ String::String(const __FlashStringHelper *pstr)
 	*this = pstr;
 }
 
-#if __cplusplus >= 201103L || defined(__GXX_EXPERIMENTAL_CXX0X__)
+#if STRING_HAS_MOVE
 String::String(String &&rval)
+	: buffer(rval.buffer)
+	, capacity(rval.capacity)
+	, len(rval.len)
```

Comment:
Really these should all have a call to `std::move`, but of course `std::move` is not available on the arduino.

Perhaps it would be sensible to provide `std::move` and `std::forward` if arduino is going to start using rvalue references internally